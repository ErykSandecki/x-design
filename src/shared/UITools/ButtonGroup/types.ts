// types
import { TTooltipProps } from '../../UI/Tooltip/Tooltip';
import { TIconProps } from '../../UI/Icon/Icon';

export type TButtonGroup = {
  disabled?: boolean;
  name: TIconProps['name'];
  onClick: TFunc;
  tooltip?: Omit<TTooltipProps, 'children'>;
};
