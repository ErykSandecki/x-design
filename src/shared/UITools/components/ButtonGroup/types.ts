// types
import { TTooltipProps } from '../../../UI/components/Tooltip/Tooltip';
import { TIconProps } from '../../../UI/components/Icon/Icon';

export type TButtonGroup = {
  disabled?: boolean;
  name: TIconProps['name'];
  onClick: TFunc;
  tooltip?: Omit<TTooltipProps, 'children'>;
};
