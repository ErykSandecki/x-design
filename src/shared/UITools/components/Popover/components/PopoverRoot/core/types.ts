export type TContext = {
  setSelected: TFunc<[boolean]>;
};
