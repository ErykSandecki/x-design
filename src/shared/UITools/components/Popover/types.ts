// types
import { TIconProps } from 'shared/UI/components/Icon/Icon';

export type TPopoverData = {
  icon?: TIconProps['name'];
  onClick?: () => void;
  selected?: boolean;
  text: string;
};

export type TPopoverSeparator = { separator: boolean };

export type TPopoverItem = TPopoverData | TPopoverSeparator;

export type TPopover = {
  data: Array<TPopoverData | TPopoverSeparator>;
};
