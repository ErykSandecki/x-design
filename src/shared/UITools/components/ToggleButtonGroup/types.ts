import { TIconProps } from 'shared/UI/components/Icon/Icon';
import { TToggleButtonProps } from './components/ToggleButton/ToggleButton';

export type TToggleButton<V> = Pick<TToggleButtonProps<V>, 'tooltip' | 'value'> & {
  icon: TIconProps['name'];
};

export type TToggleButtonGroupValue = string | Array<string>;
