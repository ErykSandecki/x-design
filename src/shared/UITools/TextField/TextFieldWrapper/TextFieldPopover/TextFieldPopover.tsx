import { FC, RefObject } from 'react';

// components
import Box from '../../../../UI/Box/Box';
import Icon from '../../../../UI/Icon/Icon';
import Popover, { PopoverCompound, TPopoverProps } from '../../../Popover/Popover';

// hooks
import { useTheme } from 'hooks';

// others
import { className, classNames } from './classNames';

// styles
import styles from './text-field-popover.scss';
import { useClickEvent } from './hooks/useClickEvent';

export type TTextFieldPopoverProps = Pick<TPopoverProps, 'children' | 'offset' | 'style'> & {
  attachedValue: boolean;
  classNameIcon: string;
  onClick: TFunc;
  ref: RefObject<HTMLDivElement>;
  selected: boolean;
  setSelected: TFunc<[boolean]>;
};

export const TextFieldPopover: FC<TTextFieldPopoverProps> = ({
  attachedValue,
  children,
  classNameIcon,
  offset,
  onClick,
  ref,
  selected,
  setSelected,
  style,
}) => {
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);
  const onClickDetachedValue = useClickEvent(onClick);

  return (
    <Box
      classes={{ className: cx(classNamesWithTheme[className]) }}
      ref={ref}
      sx={{ alignItems: 'center', display: 'flex', justifyContent: 'center', position: 'relative' }}
    >
      {!attachedValue && (
        <Icon
          classes={{ className: cx(classNamesWithTheme.icon, classNameIcon) }}
          clickable
          height={12}
          name="Variant"
          onClick={() => setSelected(!selected)}
          width={12}
        />
      )}
      {attachedValue && (
        <Icon
          classes={{ className: cx(classNamesWithTheme.icon, classNameIcon) }}
          clickable
          height={12}
          name="Detached"
          onClick={onClickDetachedValue}
          width={12}
        />
      )}
      <Popover e2eValue="popover" offset={offset} refItem={ref} selected={selected} style={style}>
        <PopoverCompound.PopoverRoot selected={selected} setSelected={setSelected}>
          {children}
        </PopoverCompound.PopoverRoot>
      </Popover>
    </Box>
  );
};

export default TextFieldPopover;
