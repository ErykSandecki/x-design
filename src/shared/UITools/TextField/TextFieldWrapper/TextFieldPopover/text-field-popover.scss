.TextFieldPopover {
  &__icon-variant {
    opacity: 0;
  }
}
