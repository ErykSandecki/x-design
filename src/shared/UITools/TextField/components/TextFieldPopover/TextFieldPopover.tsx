import { FC, HTMLAttributes, useRef } from 'react';
import { noop } from 'lodash';

// components
import Box from '../../../../UI/Box/Box';
import Icon from '../../../../UI/Icon/Icon';
import Popover, { PopoverCompound, TPopoverProps } from '../../../Popover/Popover';

// hooks
import { useOutsideClick, useTheme } from 'hooks';

// others
import { className, classNames } from './classNames';

// styles
import styles from './text-field-popover.scss';

export type TTextFieldPopoverProps = {
  children: TPopoverProps['children'];
  classNameIcon: string;
  idContainer: HTMLAttributes<any>['id'];
};

export const TextFieldPopover: FC<TTextFieldPopoverProps> = ({ children, classNameIcon, idContainer }) => {
  const ref = useRef(null);
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);
  const { selected, setSelected } = useOutsideClick([], ref, noop, idContainer);

  return (
    <Box
      classes={{ className: cx(classNamesWithTheme[className]) }}
      ref={ref}
      sx={{ alignItems: 'center', display: 'flex', justifyContent: 'center', position: 'relative' }}
    >
      <Icon
        classes={{ className: cx(classNamesWithTheme.icon, classNameIcon) }}
        clickable
        height={12}
        name="Variant"
        onClick={() => setSelected(!selected)}
        width={12}
      />
      <Popover e2eValue="popover" refItem={ref} selected={selected}>
        <PopoverCompound.PopoverRoot setSelected={setSelected}>{children}</PopoverCompound.PopoverRoot>
      </Popover>
    </Box>
  );
};

export default TextFieldPopover;
