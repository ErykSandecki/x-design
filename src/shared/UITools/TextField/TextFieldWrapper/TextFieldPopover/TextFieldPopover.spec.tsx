import { fireEvent } from '@testing-library/react';
import { RefObject } from 'react';

// components
import TextFieldPopover from './TextFieldPopover';

// types
import { E2EAttribute } from 'types';

// utils
import { createHtmlElement } from 'utils';
import { customRender, getByE2EAttribute } from 'test';

const id = 'id';
const element = createHtmlElement('div', { id });
const mockCallBack = jest.fn();
const ref = { current: null } as unknown as RefObject<HTMLInputElement>;

describe('TextFieldPopover snapshots', () => {
  beforeAll(() => {
    // mock
    document.body.appendChild(element);
  });

  it('should render TextFieldPopover', () => {
    // before
    const { asFragment } = customRender(
      <TextFieldPopover
        attachedValue={false}
        classNameIcon="classNameIcon"
        onClick={mockCallBack}
        ref={ref}
        selected={false}
        setSelected={mockCallBack}
      >
        children
      </TextFieldPopover>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render when attached value', () => {
    // before
    const { asFragment } = customRender(
      <TextFieldPopover
        attachedValue
        classNameIcon="classNameIcon"
        onClick={mockCallBack}
        ref={ref}
        selected={false}
        setSelected={mockCallBack}
      >
        children
      </TextFieldPopover>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with opened popover', () => {
    // before
    const { asFragment, container } = customRender(
      <TextFieldPopover
        attachedValue={false}
        classNameIcon="classNameIcon"
        onClick={mockCallBack}
        ref={ref}
        selected={true}
        setSelected={mockCallBack}
      >
        children
      </TextFieldPopover>,
    );

    // find
    const icon = getByE2EAttribute(container, E2EAttribute.icon, 'variant');

    // action
    fireEvent.click(icon);

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
