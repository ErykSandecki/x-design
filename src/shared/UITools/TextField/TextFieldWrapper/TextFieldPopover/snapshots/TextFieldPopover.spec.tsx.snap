// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFieldPopover snapshots should render TextFieldPopover 1`] = `
<DocumentFragment>
  <div
    class="css-zcp5dz TextFieldPopover TextFieldPopover--light"
    data-test-box=""
  >
    <div
      class="css-1jnxvbp TextFieldPopover__icon TextFieldPopover__icon--light classNameIcon Icon Icon--light Icon--clickable Icon--clickable--light"
      data-test-icon="variant"
      height="12"
      name="Variant"
      width="12"
    />
    <div
      class="css-1c8p9kb Popover Popover--light"
      data-test-box=""
      data-test-popover="popover"
      style="left: 0px; top: 0px;"
    >
      children
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextFieldPopover snapshots should render when attached value 1`] = `
<DocumentFragment>
  <div
    class="css-zcp5dz TextFieldPopover TextFieldPopover--light"
    data-test-box=""
  >
    <div
      class="css-1jnxvbp TextFieldPopover__icon TextFieldPopover__icon--light classNameIcon Icon Icon--light Icon--clickable Icon--clickable--light"
      data-test-icon="detached"
      height="12"
      name="Detached"
      width="12"
    />
    <div
      class="css-1c8p9kb Popover Popover--light"
      data-test-box=""
      data-test-popover="popover"
      style="left: 0px; top: 0px;"
    >
      children
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextFieldPopover snapshots should render with opened popover 1`] = `
<DocumentFragment>
  <div
    class="css-zcp5dz TextFieldPopover TextFieldPopover--light"
    data-test-box=""
  >
    <div
      class="css-1jnxvbp TextFieldPopover__icon TextFieldPopover__icon--light classNameIcon Icon Icon--light Icon--clickable Icon--clickable--light"
      data-test-icon="variant"
      height="12"
      name="Variant"
      width="12"
    />
    <div
      class="css-1c8p9kb Popover Popover--light Popover--selected Popover--selected--light"
      data-test-box=""
      data-test-popover="popover"
      style="left: 0px; top: 0px;"
    >
      children
    </div>
  </div>
</DocumentFragment>
`;
